doc/develop/test/ztest.rst

```diff
@@ -110,6 +110,35 @@ This is achieved via fixtures in the following way::
 Advanced features
 *****************
 
+Test result expectations
+========================
+
+Some tests were made to be broken. In cases where the test is expected to fail or skip due to the
+nature of the code, it's possible to annotate the test as such. For example::
+
+  .. code-block:: C
+
+  #include <zephyr/ztest.h>
+
+  ZTEST_SUITE(my_suite, NULL, NULL, NULL, NULL, NULL);
+
+  ZTEST_EXPECT_FAIL(my_suite, test_fail)
```

Comment:
Can we separate the two API changes as their own commits? It'll be a bit easier to review and understand which change is related to what API?